### ISSUE TITLE
Kubernetes is stuck starting

### ISSUE THREAD

**Issue by gee4vee** [docker/for-mac#5305]
- [X] I have tried with the latest version of Docker Desktop
  - [X] I have tried disabling enabled experimental features
  - [X] I have uploaded Diagnostics
  - Diagnostics ID: 6BD3A618-D74C-463B-BE5A-81C61233F16E/20210129184718

### Expected behavior
Kubernetes starts

### Actual behavior
Kubernetes is stuck in the 'starting' state.

### Information

  - macOS Version: 11.1

I have not tried enabling Kubernetes before with this installation.

### Steps to reproduce the behavior

  1. Start Docker Desktop for Mac
  2. Enable Kubernetes in Preferences.
 
I have also tried deleting the `~/Library/Group\ Containers/group.com.docker/pki` folder and restarting Docker Desktop but that didn't help.

**Comment by sayjeyhi** [comment]
I also have the same problem with MacBook with M1 chip.
<img width="290" alt="Screen Shot 1399-11-10 at 16 04 12" src="https://user-images.githubusercontent.com/6254009/106319494-c67aac00-6286-11eb-9bec-7b9c043d1596.png">

**Comment by ghishadow** [comment]
got same error too, In the end I have to do `Reset to factory defaults`, but got same error after enabling k8s again, there is #2990 #5027 with more discussion on this

**Comment by stephen-turner** [comment]
@sayjeyhi Kubernetes doesn't work on the latest M1 preview. Please see https://docs.docker.com/docker-for-mac/apple-m1/

**Comment by sayjeyhi** [comment]
@stephen-turner is there any estimation about how we could expect the k8s release for M1?

**Comment by stephen-turner** [comment]
We are expecting a preview that includes Kubernetes support soon, possibly even this week. Please remember that it's still only a preview, we haven't announced a date for a GA release yet.

**Comment by gee4vee** [comment]
FYI my MacBook is intel chip. Don’t mean to be rude but this issue was never about the M1 chip. Any word on what is going on here? I still can’t get it unstuck. I can’t even turn off Kube.

**Comment by stephen-turner** [comment]
I realise the original issue wasn't about M1. This can happen when there is corporate security software blocking the network connection to k8s.gcr.io: is that the case for you?

**Comment by gee4vee** [comment]
I’m WFH. I’ll check but doubt it.

**Comment by gee4vee** [comment]
No problem accessing that container registry.

**Comment by gee4vee** [comment]
Is there a way at least for now to turn off Kubernetes? I've tried the Reset Kubernetes button and restarting Docker Desktop. As soon as it comes up it tries to start Kubernetes and gets stuck. This is using up considerable CPU cycles.

**Comment by stephen-turner** [comment]
Can you get as far as the settings to turn it off? If not you can edit your settings.json file.

**Comment by gee4vee** [comment]
I can look into that but any update on what's causing this? I did provide my diagnostic ID.

**Comment by gee4vee** [comment]
Where can I find the settings.json file on a Mac?

**Comment by ghishadow** [comment]
click the docker icon, you will Preferences, then go to Docker Engine 
<img width="1251" alt="Screenshot 2021-02-03 at 4 52 19 PM" src="https://user-images.githubusercontent.com/9583775/106740456-7150e880-6640-11eb-9005-8fd97a6d1ebc.png">

or https://docs.docker.com/config/daemon/#configure-the-docker-daemon

**Comment by stephen-turner** [comment]
@ghishadow No, not that one. That's the engine config file. I'm talking about the Docker Desktop settings.json. Sorry, I don't remember the exact path on a Mac (I'm on Windows) but I think it's easy to find.

**Comment by jayyeras** [comment]
Same issue for me and I've tried all troubleshooting steps above including complete uninstall/reinstall. 

**Diagnostics ID:**  `EC6756B5-546C-43AD-8A6C-9D0BE43563F6/20210203182551`

**Comment by ssdowd** [comment]
On the mac, settings.json is in `~/﻿Library/Group Containers/group.com.docker`

**Comment by jayyeras** [comment]
@ssdowd This solves disabling k8s, but my issue is I need to start it and it's is stuck in starting. I've tried reseting cluster, clean/purge data, reset to factory defaults, uninstall/reinstall...

**Comment by ssdowd** [comment]
@jayyeras Correct, that is only to disable it when you cannot do that from the UI.  k8s is not yet working on apple silicon.  The original issue references intel, so resetting may only work there.

**Comment by gee4vee** [comment]
FYI resetting is not helping on Intel. Still need help here.

**Comment by sayjeyhi** [comment]
@stephen-turner any news of MacBook M1 new release?

**Comment by gee4vee** [comment]
Anyone looking at this? Any updates on a solution?

**Comment by johnlabarge** [comment]
Same here.  Stuck starting on Mac OS X 2.4 GHz Intel Core i9.

**Comment by kei95** [comment]
I got a workaround with this comment, I think it's worth trying: https://github.com/docker/for-mac/issues/5027#issuecomment-718076014

**Comment by gee4vee** [comment]
Workaround doesnt work for me and others. Any update on this?

**Comment by codeclown** [comment]
I'm running into a similar issue (stuck in Starting-state after I tried to reset the cluster from the Settings). Posted some log output here: https://github.com/docker/for-mac/issues/2990#issuecomment-808921535

**Comment by kerberjg** [comment]
@codeclown This fixed it for me, thank you! ✨
